src/util/sighash.rs

```diff
@@ -329,17 +331,28 @@ impl SchnorrSighashType {
     }
 
     /// Creates a [`SchnorrSighashType`] from raw `u8`.
+    #[deprecated(since = "0.29.0", note = "use try_from instead")]
     pub fn from_u8(hash_ty: u8) -> Result<Self, Error> {
-        match hash_ty {
-            0x00 => Ok(SchnorrSighashType::Default),
-            0x01 => Ok(SchnorrSighashType::All),
-            0x02 => Ok(SchnorrSighashType::None),
-            0x03 => Ok(SchnorrSighashType::Single),
-            0x81 => Ok(SchnorrSighashType::AllPlusAnyoneCanPay),
-            0x82 => Ok(SchnorrSighashType::NonePlusAnyoneCanPay),
-            0x83 => Ok(SchnorrSighashType::SinglePlusAnyoneCanPay),
-            x => Err(Error::InvalidSighashType(x as u32)),
-        }
+        Self::try_from(hash_ty)
+    }
+}
+
+impl TryFrom<u8> for SchnorrSighashType {
+    type Error = Error;
+
+    fn try_from(hash_ty: u8) -> Result<Self, Self::Error> {
+        use SchnorrSighashType::*;
+
+        Ok(match hash_ty {
+            0x00 => Default,
+            0x01 => All,
+            0x02 => None,
+            0x03 => Single,
+            0x81 => AllPlusAnyoneCanPay,
+            0x82 => NonePlusAnyoneCanPay,
+            0x83 => SinglePlusAnyoneCanPay,
+            x => return Err(Error::InvalidSighashType(x as u32)),
+        })
```

Comment:
Shouldn't this be `from_consensus_u8`?

Reply:
I do not know how Bitcoin Core handles a a schnorr sighash type that is _not_  0-3 or 81-83. Looking solely at our code, for `EcdsaSighashType` we have `from_standard` and `from_consensus`, the first returns an error the later is infallible. If we are to use consensus/standard as part of the identifier for `SchnorrSighashType` should it not be the same, or does schnorr sighash not have this concept of standard vs consensus?

Reply:
Any sighash type that is not 0-3 or 81-83 is rejected by consensus rules.

Reply:
Ok, two opinions - check!

Will re-name to `from_consensus_u8`, thanks.